Warsaw_Tram/InitialViewController.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  ViewController.swift
+//  Warsaw_Tram
+//
+//  Created by Małgorzata Dziubich on 23/06/16.
+//  Copyright © 2016 Małgorzata Dziubich. All rights reserved.
+//
+
+import UIKit
+
+class InitialViewController: UIViewController, AlertHelperProtocol {
+
+    @IBOutlet var pickerView: UIPickerView!
+    @IBOutlet var lowFloorTramSwitch: UISwitch!
+    
+    let tramViewModel = TramViewModel()
+    
+    var allActiveTrams = [Tram]()
+    var tramsNumberList = [Int]()
+    var lowFloorTramsNumberList = [Int]()
+    var tramNumberToDisplayOnMap = String()
+    var showAllTrams: Bool = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        lowFloorTramSwitch.on = false
+        fetchTramsData()
+    }
+    
+    @IBAction func showTramsButton(sender: AnyObject) {
+        if tramNumberToDisplayOnMap == "" {
+            showError(message: "You need to choose a tram!")
+        } else {
+            self.performSegueWithIdentifier("ShowMap", sender: self)
+        }
+    }
+    
+    @IBAction func showAllTramsButton(sender: AnyObject) {
+        showAllTrams = true
+        self.performSegueWithIdentifier("ShowMap", sender: self)
+    }
+    
+    @IBAction func switchValueDidChange(sender: AnyObject) {
+        pickerView.reloadAllComponents()
+    }
+    
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        let viewController = segue.destinationViewController as! MapViewController
+        viewController.allActiveTrams = allActiveTrams
+        viewController.tramNumberToDisplayOnMap = tramNumberToDisplayOnMap
+        viewController.lowFloorFilter = lowFloorTramSwitch.on
+        viewController.showAllTrams = showAllTrams
```

Comment:
Setting so many values doesn't looks good. I suggest to create a `struct` that will be responsible for store all data need to properly display map and pass only this struct to `MapViewController`